cpp_utils/test/unittest/qos/qos_utils_test.cpp

```diff
@@ -0,0 +1,402 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cpp_utils/testing/gtest_aux.hpp>
+#include <gtest/gtest.h>
+
+#include <fastdds/dds/core/policy/QosPolicies.hpp>
+
+#include <cpp_utils/qos/qos_utils.hpp>
+
+using namespace eprosima;
+using namespace eprosima::fastdds::dds;
+
+/**
+ * Test \c qos_policy_id_to_string call
+ */
+TEST(qos_utils_test, qos_policy_id_to_string)
+{
+    // User data
+    {
+        const auto id = QosPolicyId_t::USERDATA_QOS_POLICY_ID;
+        const auto id_str = utils::qos_policy_id_to_string(id);
+        const auto id_str_expected = "User data";
+        ASSERT_EQ(id_str, id_str_expected);
```

Comment:
Try to parametrize this test, you are writing the same steps to test all `QosPolicyId_t`, why not creating a `std::map<uint32_t, std::string>` (with every `QosPolicyId_t` and its corresponding `std::string`) and iterating through it?